azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAppRoleAuthenticationProperties.java

```diff
@@ -0,0 +1,15 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.autoconfigure.aad;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
+
+@Validated
+@ConfigurationProperties("azure.aad.app-role")
```

Comment:
please follow the naming conversion as AADAuthenticationProperties, use `azure.activedirectory` instead of `azure.aad.app-role`.

Reply:
The property "namespace" `azure.activedirectory` is already bound to `com.microsoft.azure.spring.autoconfigure.aad.AADAuthenticationProperties`. I did not manage to use `azure.activedirectory.app-role` for my new configuration without having side effects of `AADAuthenticationProperties` being instantiated and beans being created automatically. As I didn't want to break the existing feature I've chosen this different property name. 
Is there a way of doing it without the side effects I've described?